### PR TITLE
[Merged by Bors] - ET-3868 Build test config for mind benchmarks

### PR DESCRIPTION
**Pull request by PanicButtonPressed**
**Reference**

[ET-3868](https://xainag.atlassian.net/browse/ET-3868)

**Summary**

This PR introduces a test config including the configs for coi system and personalization which can be passed to the `State` struct.

[ET-3868]: https://xainag.atlassian.net/browse/ET-3868?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: web-api/src/mind.rs [hunk 1]
```diff
@@ -187,6 +194,12 @@ impl Default for PersonaBasedConfig {
     }
 }
 
+#[derive(Debug)]
+struct TestConfig {
+    coi_config: CoiConfig,
+    personalization_config: PersonalizationConfig,
```
Comment (janpetschexain) [inline web-api/src/mind.rs:200]: :art: i'd rather call it `StateConfig` as it is used as the config for the `State`, wdyt?

:art: we can also make the field names less redundant, they are already part of a config type.

```suggestion
struct StateConfig {
    coi: CoiConfig,
    personalization: PersonalizationConfig,
```

File: coi/src/config.rs [hunk 2]
```diff
@@ -28,16 +28,16 @@ use crate::{
 #[must_use]
 pub struct Config {
     #[serde(default = "default_shift_factor")]
-    shift_factor: f32,
+    pub shift_factor: f32,
```
[the remark this answers is not in the document]
Comment (PanicButtonPressed) [reply coi/src/config.rs:31]: makes sense

File: web-api/src/mind.rs (hunk 1 above)
Comment (PanicButtonPressed) [reply web-api/src/mind.rs:200]: agree

File: web-api/src/mind.rs [hunk 3]
```diff
@@ -54,13 +54,20 @@ struct State {
 }
 
 impl State {
-    fn new(storage: Storage) -> Result<Self, Error> {
+    fn new(storage: Storage, test_config: Option<TestConfig>) -> Result<Self, Error> {
```
[the remark this answers is not in the document]
Comment (PanicButtonPressed) [reply web-api/src/mind.rs:57]: indeed pretty nice use of `default` trait

**Review by janpetschexain — APPROVED** [review]
lgtm

**Comment by PanicButtonPressed** [comment]
bors merge

**Comment by PanicButtonPressed** [comment]
bors merge

**Pull request closed without merging**